### PR TITLE
disable unnecessary CMake build options

### PR DESCRIPTION
**Pull request by kostmo**
With these changes, I get the following error when running `cmake ..`:

```
CMake Error at cmake/Codegen.cmake:163 (message):
  Failed to get generated_cpp list
Call Stack (most recent call first):
  caffe2/CMakeLists.txt:2 (include)
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,12 +87,12 @@ option(BUILD_ATEN_MOBILE "Build ATen for Android and iOS" OFF)
...
 option(BUILD_DOCS "Build Caffe2 documentation" OFF)
-option(BUILD_CUSTOM_PROTOBUF "Build and use Caffe2's own protobuf under third_party" ON)
-option(BUILD_PYTHON "Build Python binaries" ON)
-option(BUILD_CAFFE2_OPS "Build Caffe2 operators" ON)
+option(BUILD_CUSTOM_PROTOBUF "Build and use Caffe2's own protobuf under third_party" OFF)
+option(BUILD_PYTHON "Build Python binaries" OFF)
+option(BUILD_CAFFE2_OPS "Build Caffe2 operators" OFF)
 option(BUILD_SHARED_LIBS "Build libcaffe2.so" ON)
 cmake_dependent_option(
-    CAFFE2_LINK_LOCAL_PROTOBUF "If set, build protobuf inside libcaffe2.so." ON
```
Comment (pjh5) [inline CMakeLists.txt]: Keep this one on

File: CMakeLists.txt [hunk 2]
```diff
@@ -87,12 +87,12 @@ option(BUILD_ATEN_MOBILE "Build ATen for Android and iOS" OFF)
 option(BUILD_ATEN_ONLY "Build only a subset focused on ATen only" OFF)
 option(BUILD_BINARY "Build C++ binaries" OFF)
 option(BUILD_DOCS "Build Caffe2 documentation" OFF)
-option(BUILD_CUSTOM_PROTOBUF "Build and use Caffe2's own protobuf under third_party" ON)
```
Comment (pjh5) [inline CMakeLists.txt]: Keep this one on

**Comment by kostmo** [comment]
Here's the reduction in the CMake dependency tree:

Before:
deps-full
After:
deps-partial

**Review by soumith — COMMENTED** [review]
is the idea of this PR that we set these to off, but build_pytorch_libs.py controls the default values?

For example USE_CUDNN we'd definitely want ON ...

**Comment by kostmo** [comment]
This is strictly for my own development speed and will not be merged.

**Comment by kostmo** [comment]
FYI this subset of options reduces my local (4-core Mac) compile time from 18min to 7min.

**Pull request closed without merging**